nwb_conversion_tools/datainterfaces/ecephys/axona/axonadatainterface.py

```diff
@@ -124,121 +135,233 @@ def get_metadata(self):
+        self.recording_extractor = se.AxonaUnitRecordingExtractor(filename=filename, noise_std=noise_std)
+
+
 # Helper functions for AxonaPositionDataInterface
-def establish_mmap_to_position_data(filename):
+def get_header_bstring(file):
     """
-    Generates a memory map (mmap) object connected to an Axona .bin
-    file, referencing only the animal position data (if present).
+    Scan file for the occurrence of 'data_start' and return the header
+    as byte string
 
-    When no .bin file is available or no position data is included,
-    returns None.
+    Parameters
+    ----------
+    file (str or path): file to be loaded
 
-    TODO: Also allow using .pos file (currently only support .bin)
+    Returns
+    -------
+    str: header byte content
+    """
+    header = b""
+    with open(file, "rb") as f:
+        for bin_line in f:
+            if b"data_start" in bin_line:
+                header += b"data_start"
+                break
+            else:
+                header += bin_line
+    return header
+
+
+def read_bin_file_position_data(bin_filename):
+    """
+    Read position data from Axona `.bin` file (if present).
 
     Parameters:
     -------
-    filename (Path or Str): Full filename of Axona file with any
-        extension.
+    bin_filename (Path or Str):
+        Full filename of Axona file with any extension.
 
     Returns:
     -------
-    mm (mmap or None): Memory map to .bin file position data
+    np.array
+        Columns are time (ms), X, Y, x, y, PX, px, tot_px, unused
+
+    Notes:
+    ------
+    To obtain the correct column order we pairwise flip the 8 int16 columns
+    described in the file format manual. In addition, note that `.bin` data is
+    little endian (read right to left), as opposed to `.pos` file data, which is
+    big endian.
     """
-    mmpos = None
-
-    bin_file = filename.split(".")[0] + ".bin"
-    set_file = filename.split(".")[0] + ".set"
-    par = parse_generic_header(set_file, ["rawRate", "duration"])
-    sr_ecephys = int(par["rawRate"])
-    sr_pos = 100
-    bytes_packet = 432
-
-    num_packets = int(os.path.getsize(bin_file) / bytes_packet)
-    num_ecephys_samples = num_packets * 3
-    dur_ecephys = num_ecephys_samples / sr_ecephys
-    assert dur_ecephys == float(par["duration"])
-
-    # Check if position data exists in .bin file
-    with open(bin_file, "rb") as f:
-        with contextlib.closing(
-            mmap.mmap(
-                f.fileno(),
-                sr_ecephys // 3 // sr_pos * bytes_packet,
-                access=mmap.ACCESS_READ,
-            )
-        ) as mmap_obj:
-            contains_pos_tracking = mmap_obj.find(b"ADU2") > -1
+    pos_dt_se = np.dtype(
+        [
+            ("t", "<i4"),
+            ("X", "<i2"),
+            ("Y", "<i2"),
+            ("x", "<i2"),
+            ("y", "<i2"),
+            ("PX", "<i2"),
+            ("px", "<i2"),
+            ("tot_px", "<i2"),
+            ("unused", "<i2"),
+        ]
+    )
```

Comment:
cool use of a custom hierarchical datatype! I haven't seen memmaps used this way before

Reply:
Shout out to @alejoe91 , I did not know this before he showed me either  ;-)